### ISSUE TITLE
Be more flexible about local h2 databases to allow distributing read-only copies for offline maven builds.

### ISSUE THREAD

**Issue by trandersen-ufst** [dependency-check/DependencyCheck#4383]
**Is your feature request related to a problem? Please describe.**

We are looking into pre-creating the databases used by the dependency check maven plugin to allow being run as part of an offline CICD maven build.

My idea so far is to have a SNAPSHOT build which uses src/main/resources as the datafolder (instead of sneaking in a corner of .m2) which then by default puts the ovc.mv.db file in the root of the generated jar file which is then in turn managed completely by Maven simply by installing it and then use it as a dependency and then tell the plugin to use a "jdbc:h2:zip:..." url instead.

Unfortunately the plugin logic explicitly checks in isH2Connection if the prefix is "jdbc:h2:file" so this breaks.

If I can get this to work, there will be another way of running the plugin in a locked down environment which will be reasonably fast and not require having a database server running.

Is this something that would be interesting to the project?

**Comment by aikebah** [comment]
First thing that springs to mind is: why would you create such a complex setup that requires a mvn install? Why not simply use the datadir to get it out of your .m2 repo to some stable location?

**Comment by aikebah** [comment]
With the database in a zip-file you will suffer performance-wise per [H2 documentation](http://www.h2database.com/html/features.html?highlight=zip&search=zip#database_in_zip).
Also note that the isH2Connection is only used during the database update scenarios (to trigger defragmentation of the database after the updates and to create the database schema if the H2 database was not yet present), which are not possible anyhow for an H2 database in a zip-file.
So for your scenario the isH2Connection can safely return false eventhough the JDBC URL you configure is in fact a read-only h2 connection.

**Comment by trandersen-ufst** [comment]
@aikebah Thank you for your comments.

The basic problem is that the build servers that need to run `mvn dependency-check:check` as part of building whatever we want it to, do not have internet access and cannot reach the official servers.  Also they run on an empty .m2 folder, so the work to download and construct the database needs to be done in every run as they are not there, which takes several minutes at best, and the documentation does not cover this particular case.   A way around this is to provide data files from a previous run.  Unfortunately previous runs did not have access to the internet either, so they have to be brought in from a job running outside the build server enclosure.   I was looking into using Maven 

Since I wrote this I found that it is not enough to just have the database file available with a JDBC connection, the whole set of data files are needed, which makes this approach somewhat moot.

I ended up so far with using the `src/main/resources` as the data dir for the online building step, and then deploy the artifact containing the datafiles to an Artifactory server visible from the inside of the offline CICD building enclosure, and run dependency:unpack to a folder in target/ just before running dependency-check:check with AutoUpdates disabled.   This has worked for our proof-of-concept.

That in turn means that we do not need anything from you to get this to run, and I'll therefore close this issue.
